### PR TITLE
feat: 🎸 add `removeEmptyValue` config

### PR DESCRIPTION
**Pull request by ritox842**
Previously the url contained all synced form values - including empty
values.

BREAKING CHANGE: 🧨 From now on, empty control values will be removed from the URL query
params.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [V] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [V] Tests for the changes have been added (for bug fixes / features)
- [V ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[V] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently, empty control values are syncing to the URL query params causing the URL to contain keys without any values which is messy and in some cases unwanted.

Issue Number: N/A

## What is the new behavior?

From now on, By default, Any control empty value will be removed from the URL query params.
This behavior can be changed back to the previous way when setting `removeEmptyValue` to false in the def config.

## Does this PR introduce a breaking change?

```
[V] Yes
[ ] No
```

From now on, empty control values will be removed from the URL query params.

### CODE REVIEW

File: projects/ngneat/bind-query-params/src/lib/utils.ts [hunk 1]
```diff
@@ -1,6 +1,19 @@
 import { AbstractControl } from '@angular/forms';
 import { ParamDefType, ResolveParamsOption } from './types';
 import { QueryParamDef } from './QueryParamDef';
+import size from 'lodash.size';
+
+function isEmptyValue(def: QueryParamDef, value: any) {
```
Comment (NetanelBasal) [inline projects/ngneat/bind-query-params/src/lib/utils.ts:6]: I prefer not to include an additional package for this. We can do the following:

```ts
function isEmptyValue(value: any) {
	if(Array.isArray(value)) {
	   return !!value.length 
	}

  if(value !== null && typeof value === 'object'){
     return Object.keys(value).length;
  }

  return !!value;
}
```

**Review by NetanelBasal — COMMENTED** [review]
Great PR

File: projects/ngneat/bind-query-params/src/lib/utils.ts (hunk 1 above)
Comment (ritox842) [reply projects/ngneat/bind-query-params/src/lib/utils.ts:6]: You don't think that an empty string should be removed from the URL?

What about 
```
function isEmptyValue(def: QueryParamDef, value: any) {
  switch (def.type) {
    case 'array':
      return !!value.length;
    case 'object':
      return !!Object.keys(value).length;
    case 'string':
      return !value || value === '';
    default:
      return false;
  }
}
```

Comment (NetanelBasal) [reply projects/ngneat/bind-query-params/src/lib/utils.ts:6]: My code handles empty strings, but I see that we know the type in advance, so it's better to use your version.

Comment (ritox842) [reply projects/ngneat/bind-query-params/src/lib/utils.ts:5]: 👍 

I'm updating my PR with this method and removing lodash.size completely.

**Pull request merged**